### PR TITLE
fix: fixed the logical_not not support the bfloat16

### LINKED ISSUE #28454: Fix `ivy.logical_not` support for bfloat16 at `paddle backend`

### PR DESCRIPTION
**Pull request by samthakur587**
# fixed the logical_not not support the bfloat16 at paddle backend 

## fix support for bfloat16 at paddle backend that the paddle doesn't support the bfloat16 dtype and removed the `python if paddle.is_complex(x):
        return paddle.logical_and(
            paddle.logical_not(x.real()), paddle.logical_not(x.imag())
        ) # the paddle itself support the complex dtype for paddle.logical_not `

passing all tests at local.

image

Closes #28454

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

File: ivy/functional/backends/paddle/elementwise.py [hunk 1]
```diff
@@ -377,14 +377,12 @@ def cos(x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None) -> paddle.T
...
 
 
-@with_unsupported_dtypes({"2.6.0 and below": ("uint8", "float16")}, backend_version)
+@with_unsupported_dtypes(
+    {"2.6.0 and below": ("uint8", "float16", "bfloat16")}, backend_version
+)
 def logical_not(
     x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None
 ) -> paddle.Tensor:
-    if paddle.is_complex(x):
```
Comment (Ishticode) [inline ivy/functional/backends/paddle/elementwise.py:384]: we seem to be removing the complex dtype handling. So is complex unsupported now or does it still work?

**Review by Ishticode — COMMENTED** [review]
Hi Sam. Thanks for the PR.

File: ivy/functional/backends/paddle/elementwise.py (hunk 1 above)
Comment (samthakur587) [reply ivy/functional/backends/paddle/elementwise.py:384]: yes it's working bcz the paddle.logical_not can handle the complex dtype also.  [here](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/logical_not_en.html) the paddle doc for this and i have tested this passing all tests in local.

**Review by Ishticode — APPROVED** [review]
Thanks @samthakur587 
Will merge soon

**Pull request merged**